src/testdir/test_unlet_env.vim

```diff
@@ -0,0 +1,25 @@
+" Test for unlet $FOOBAR.
+
+func Test_UnletEnv()
+  let envcmd = has('win32') ? 'set' : 'env'
+
+  let $FOOBAR='test'
+  let found = 0
+  for kv in split(system(envcmd), "\r*\n")
+    if kv == 'FOOBAR=test'
+      let found = 1
+    endif
+  endfor
+  call assert_equal(1, found)
+
+  unlet $FOOBAR
+  let found = 0
+  for kv in split(system(envcmd), "\r*\n")
+    if kv == 'FOOBAR=test'
+      let found = 1
+    endif
+  endfor
+  call assert_equal(0, found)
+
+  unlet $MUST_NOT_BE_AN_ERROR
```

Comment:
You should probably also check for unexistence of this variable first. It is also not obvious from the variable name that you check for unletting unexistent variable. If writing tests for Neovim I would check that with `eq(0, exc_exec('unlet $XXX_NONEXSTENT_VAR_XXX'))`: variable name states that it is unexistent and surrounding code states that it is checking for no exceptions.

Though why not just use `unlet $FOOBAR` again?

Reply:
I'm thinking non-existence variable should not be an error since `:echo $NOT_FOUND_ENV` doesn't make an error.

Reply:
bash doesn't make it an error.

```
$ unset FOOOOOOOOOOOOOO
```

Reply:
@mattn I am not saying it should be an error. I am saying that variable *may exist* and it is good to check this is not the case first, primary to make it obvious what exactly test is checking. And/or name variable so that it is obvious that it is not supposed to exist. Or use `$FOOBAR` as it was just deleted.

Though bash `unlet FOOOO` is *not relevant*. `unset g:xxx_nonexistent_var_xxx` will throw, `unset! …` won’t and this is VimL and not bash.

Reply:
As long as non-existence variable doesn't make an error with reference, I'm thinking unlet should not make error. @brammool How do you think?